src/main/java/edu/unc/lib/boxc/migration/cdm/services/GroupMappingService.java

```diff
@@ -88,50 +88,109 @@ public void generateMapping(GroupMappingOptions options) throws IOException {
+        if (needsMerge) {
+            mergeUpdates(options, mappingPath);
+        }
+
+        if (!options.getDryRun()) {
+            setUpdatedDate(Instant.now());
+        }
+    }
+
+    private void generateOneGroupMapping(GroupMappingOptions options, Statement stmt,
+                                         HashSet<String> multiMemberGroupSet, CSVPrinter csvPrinter) throws Exception {
+        ResultSet groupRs = stmt.executeQuery("select " + options.getGroupField().get(0)
+                + " from " + CdmIndexService.TB_NAME
+                + " where " + CdmIndexService.ENTRY_TYPE_FIELD + " is null"
+                + " group by " + options.getGroupField().get(0)
+                + " having count(*) > 1");
+        while (groupRs.next()) {
+            var groupValue = groupRs.getString(1);
+            if (StringUtils.isBlank(groupValue)) {
+                continue;
+            }
+            multiMemberGroupSet.add(groupValue);
+        }
+
+        ResultSet rs = stmt.executeQuery("select " + CdmFieldInfo.CDM_ID + ", " + options.getGroupField().get(0)
+                + " from " + CdmIndexService.TB_NAME
+                + " where " + CdmIndexService.ENTRY_TYPE_FIELD + " is null"
+                + " order by " + CdmFieldInfo.CDM_ID + " ASC");
+        while (rs.next()) {
+            String cdmId = rs.getString(1);
+            String matchedValue = rs.getString(2);
+
+            // Add empty mapping for records either not in groups or in groups with fewer than 2 members
+            if (StringUtils.isBlank(matchedValue) || !multiMemberGroupSet.contains(matchedValue)) {
+                log.debug("No matching field for object {}", cdmId);
+                csvPrinter.printRecord(cdmId, null);
+                continue;
+            }
+
+            String groupKey = GroupMappingInfo.GROUPED_WORK_PREFIX + options.getGroupField().get(0) + ":" + matchedValue;
+            csvPrinter.printRecord(cdmId, groupKey);
+        }
+    }
+
+    private void generateMultipleGroupMapping(GroupMappingOptions options, Statement stmt,
+                                              HashSet<String> multiMemberGroupSet, CSVPrinter csvPrinter) throws Exception {
+        int numberGroups = options.getGroupField().size();
+        String multipleGroups = String.join(", ", options.getGroupField());
+
+        ResultSet groupRs = stmt.executeQuery("select " + multipleGroups
+                + " from " + CdmIndexService.TB_NAME
+                + " where " + CdmIndexService.ENTRY_TYPE_FIELD + " is null"
+                + " group by " + multipleGroups
+                + " having count(*) > 1");
+        while (groupRs.next()) {
+            for (int i = 1; i < numberGroups + 1; i++) {
+                var groupValue = groupRs.getString(i);
                 if (StringUtils.isBlank(groupValue)) {
                     continue;
                 }
                 multiMemberGroupSet.add(groupValue);
```

Comment:
I think the value we store in `multiMemberGroupSet` should be the full key of the group, otherwise I think we could end up with unexpected results. Like if we were grouping on `field1` and `field2`, and we have five records:
```
cdmid: 1
field1: a
field2: b

cdmid: 2
field1: a
field2: b

cdmid: 3
field1: c
field2: d

cdmid: 4
field1: c
field2: d

cdmid: 5
field1: a
field2: c
```

I think at the moment we'd end up with a group entry for cdmid 5 since down below it'd return matched fields and both of its values are in `multiMemberGroupSet`.

So I'd suggest joining all the group values (with a "," or "|" delimiter) and adding them as a single key per group in `multiMemberGroupSet`.

I think we also might need to include blank groupValues in the joined key, but we'd want to skip adding to `multiMemberGroupSet` if all the groupValues were blank. Otherwise we could run into key collisions.